cdisc_rules_engine/operations/operation_interface.py

```diff
@@ -0,0 +1,47 @@
+import pandas as pd
+from cdisc_rules_engine.models.operation_params import OperationParams
+from abc import ABC, abstractmethod
+
+from cdisc_rules_engine.services.cache.cache_service_interface import (
+    CacheServiceInterface,
+)
+from cdisc_rules_engine.services.data_services.base_data_service import BaseDataService
+
+
+class OperationInterface(ABC):
+    def __init__(
+        self,
+        params: OperationParams,
+        dataset: pd.DataFrame,
+        cache_service: CacheServiceInterface,
+        data_service: BaseDataService,
+    ):
+        self.params = params
+        self.cache = cache_service
+        self.data_service = data_service
+        self.evaluation_dataset = dataset
+
+    @abstractmethod
+    def execute(self) -> pd.DataFrame:
+        """Perform operation calculations."""
+        pass
+
+    def _handle_operation_result(self, result) -> pd.DataFrame:
```

Comment:
An interface must not contain any implementation. Please create an abstract class and move _handle_operation_result there. I have also noticed that all operations duplicate  _handle_operation_result call. I think it would be great to call this method only once.

Reply:
Would it be better just to change the operation interface class into a BaseOperation class? I don’t see a need for an interface and a base class

Reply:
I think we can. Then this base class can have a public `execute` method which will have 2 steps: execute the operation logic, handle the result. "execute the operation logic" part can be abstract and the concrete operation classes will implement it. This way, `_handle_operation_result` will be called only once.